ansible_collections/arista/avd/roles/eos_designs/templates/facts/topology/p2p-uplinks.j2

```diff
@@ -6,32 +6,34 @@ topology:
   links:
 {% for uplink_interface in switch.uplink_interfaces | arista.avd.default([]) %}
 {%     set uplink_switch = switch.uplink_switches[loop.index0] | arista.avd.default() %}
-{%     if uplink_switch is arista.avd.defined and uplink_switch in groups[fabric_name] and
-          hostvars[uplink_switch].switch is arista.avd.defined(fail_action='warning',var_name='hostvars[' ~ uplink_switch ~ '].switch') %}
-{%         do topology_peers.append(uplink_switch) %}
+{%     if uplink_switch is arista.avd.defined and uplink_switch in groups[fabric_name] %}
+{%         set uplink_switch_facts = hostvars[uplink_switch].switch | arista.avd.default %}
```

Comment:
We can set this in topology/main.j2 so that we can reuse again in port-channel-uplinks.j2 instead of creating once again there.

Reply:
We could, but these two templates don't run for the same device. The p2p-uplinks are for l3 devices like l3leaf and spines and the port-channel-uplinks is for l2leaf.

Reply:
I agree with @ClausHolbechArista, and let's focus on speed optimization only and not increase the scope of this PR.

Reply:
My bad. I didn't check the logic in main.j2, I agree this is correct logic.